compiler/src/iree/compiler/GlobalOptimization/PropagateLinalgTranspose.cpp

```diff
@@ -481,6 +765,44 @@ class GeneralizeInputTransposedNamedOp
   }
 };
 
+// Generalizes a transpose on a hoistable op, stopping propagation.
```

Comment:
Are you making this generalization so that const-hoisting/const-eval can pick these up? Wouldnt it be simpler to just allow const-eval to pick up transposes?

Reply:
All of the patterns for propagation look for a `linalg.transpose` op. By generalizing it, none of the patterns will apply anymore, thus stopping propagation for that transpose.

Reply:
Not sure why you need that though. You cant propagate this transpose through a constant or a global load anyway. So why is this even needed here?

Reply:
I just went ahead and dropped the whole pattern. We can add it back later if needed.